### ISSUE TITLE
Oracle Support

### ISSUE THREAD

**Issue by dmcghan** [lusakasa/sqorn#2]
Looks very cool! I'd love to see support for Oracle Database and bind variables.

**Comment by eejdoowad** [comment]
Is a free version of Oracle available? 

The code is structure to support multiple database libraries. For now, I'm just working on getting Postgres production ready. If someone were to submit a PR supporting another databases, I would accept it.

**Comment by dmcghan** [comment]
@eejdoowad 

> Is a free version of Oracle available?

There is, it's called [Oracle XE](https://www.oracle.com/database/technologies/appdev/xe.html). The current version is 11.2, but 18.? will hopefully be out in the near future.


> For now, I'm just working on getting Postgres production ready.

Does Postgres use bind variables? If yes, do you support them - the examples didn't look like they did. Here's the last one:

```js
// UPDATE
await Person({ id: 23 }).set({ name: 'Rob' })
// "update person where id = 23 set name = 'Rob'"
```

With bind variables, that might look like:
```js
// UPDATE
await Person({ id: 23 }).set({ name: 'Rob' })
// "update person where id = :bind_1 set name =:bind_2"
// binds = [23,  'Rob']
```

If bind variables are supported, it should be easy to add support for other databases when that work is done.

**Comment by eejdoowad** [comment]
Yes, Sqorn parameterizes variables to prevent SQL injection, so it shouldn't be too difficult. For Postgres, this looks like:

```js
await Person({ id: 23 }).set({ name: 'Rob' })
// { txt: 'update person set name = 'Rob' where id = $1', arg: [23] }
```

**Comment by cemremengu** [comment]
Nice! It would be great if you provide a short guideline on how to implement a new dialect after you finalize the structure.
